tern/__main__.py

```diff
@@ -11,8 +11,14 @@
 import argparse
 import logging
 import os
+import sys
+
+try:
+    from tern.report import report
+except Exception as e:
+    print("Couldn't set up the report module:\n\t{}".format(e))
+    sys.exit(1)
```

Comment:
This is reasonable if running in a shell but not if you're running in a detached docker container you won't see this error in a situation where you're trying to connect to a registry or the host's unix socket. Maybe replace the `print` with the `logger`? That would mean moving the logger setup below before this import.